src/isar_turtlebot/turtlebot/turtlebot.py

```diff
@@ -50,67 +37,28 @@ def __init__(self, bridge: RosBridge) -> None:
         self.task_handler: Optional[TaskHandler] = None
 
         self.filenames: dict = dict()
+        self.metadata: dict = dict()
 
     def publish_task(self, task: Task) -> None:
-        task_input: Pose
-        if isinstance(task, DriveToPose):
-            task_input = task.pose
-        elif isinstance(task, (TakeImage, TakeThermalImage)):
-            task_input = get_inspection_pose(
-                target=task.target, current_pose=self.get_robot_pose()
-            )
-
         self.task_handler = self.task_handlers[type(task).__name__]
-        self.task_handler.start(task_input)
+        self.task_handler.start(task)
+
+        if isinstance(task, InspectionTask):
+            self.filenames[task.id] = self.task_handler.get_filename()
+            self.metadata[task.id] = self.task_handler.get_inspection()
 
     def get_task_status(self) -> TaskStatus:
         if self.task_handler:
             status: Status = self.task_handler.get_status()
             return Status.map_to_task_status(status=status)
 
-    def get_robot_pose(self) -> Pose:
-        pose_message: dict = self.bridge.pose.get_value()
-        pose: Pose = decode_pose_message(pose_message=pose_message)
-        return pose
-
-    def register_inspection_id(self, inspection_id: UUID) -> None:
-        self.filenames[inspection_id] = self.task_handler.filename
+    def get_inspections(self, id: UUID) -> Sequence[Inspection]:
+        inspection: Inspection = self.metadata[id]
+        inspection.data = self._read_data(id)
```

Comment:
Consider adding error handling when retrieving inspection data, like `FileNotFoundError` during reading or `KeyError` if the inspection ID does not match any filename indices.

Reply:
Good suggestion!